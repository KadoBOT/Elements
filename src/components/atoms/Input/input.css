.input {
  flex: 1;
}

.input__label {
  margin: 10px 0px;
  position: relative;
}

.input__label > p {
  color: black;
  text-align: left;
}

.input__required {
  color: #ec0f50;
  position: absolute;
  top: -5px;
}

.input > input {
  width: 100%;
  height: 30px;
  border-radius: 3px;
  border: 1px solid #b5b5b5;
  padding: 15px 10px;
  height: 49px;
  font-size: 26px;
}
