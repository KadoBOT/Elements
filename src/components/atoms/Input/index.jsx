import React, {PropTypes} from 'react'

import { Text } from '../Text'
import './input.css'

const Input = ({isRequired, label, placeholder, type}) => {
  return(
    <label className="input">
      <p className="input__label">
        <Text>
          {label}
          {isRequired && <span className="input__required">â€¢</span>}
        </Text>
      </p>
      <input
        placeholder={placeholder}
        type={type}
      />
    </label>
  )
}

Input.propTypes = {
  props: PropTypes.type
}

export default Input
