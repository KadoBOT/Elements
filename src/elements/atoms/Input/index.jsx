import React, {PropTypes} from 'react'
import R from 'ramda'

import { Text } from '../Text'
import './input.css'

const Input = ({errors, isRequired, label, name, onChange, placeholder, type, value}) => (
  <label className="input">
    <div className="input__label">
      <Text>
        {label}
        {isRequired && <span className="input__required">â€¢</span>}
      </Text>
    </div>
    <input
      className={errors && `input__error-${errors.length !== 0}`}
      name={name}
      onChange={onChange}
      placeholder={placeholder}
      required={isRequired}
      type={type}
      value={value}
    />
    {errors && <ul className="input__errors">
      {R.map(e =>
        <li>{e}</li>
      , errors)}
    </ul>}
  </label>
)

Input.propTypes = {
  errors: PropTypes.array.isRequired,
  isRequired: PropTypes.bool,
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired
}

export default Input
