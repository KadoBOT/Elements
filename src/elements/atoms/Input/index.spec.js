import React from 'react'
import { shallow, render, mount } from 'enzyme';

import Input from './'

it('should render the Input', () => {
    const wrapper = shallow(
        <Input
          errors={[]}
          isRequired={true}
          label="name"
          name="name"
          onChange={jest.fn()}
          placeholder="place name here..."
          type="text"
          value="John Doe"
        />
    );
    expect(wrapper).toMatchSnapshot();
});
