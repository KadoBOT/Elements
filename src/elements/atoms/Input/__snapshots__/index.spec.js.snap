exports[`test should render the Input 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <label
    className="input">
    <div
        className="input__label">
        <styled.p>
            name
            <span
                className="input__required">
                •
            </span>
        </styled.p>
    </div>
    <input
        className="input__error-false"
        name="name"
        onChange={[Function]}
        placeholder="place name here..."
        required={true}
        type="text"
        value="John Doe" />
    <ul
        className="input__errors">
        
    </ul>
</label>,
  "nodes": Array [
    <label
      className="input">
      <div
            className="input__label">
            <styled.p>
                  name
                  <span
                        className="input__required">
                        •
                  </span>
            </styled.p>
      </div>
      <input
            className="input__error-false"
            name="name"
            onChange={[Function]}
            placeholder="place name here..."
            required={true}
            type="text"
            value="John Doe" />
      <ul
            className="input__errors">
            
      </ul>
</label>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown
        errors={Array []}
        isRequired={true}
        label="name"
        name="name"
        onChange={[Function]}
        placeholder="place name here..."
        type="text"
        value="John Doe" />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "errors": Array [],
          "isRequired": true,
          "label": "name",
          "name": "name",
          "onChange": [Function],
          "placeholder": "place name here...",
          "type": "text",
          "value": "John Doe",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <label
          className="input">
          <div
                    className="input__label">
                    <styled.p>
                              name
                              <span
                                        className="input__required">
                                        •
                              </span>
                    </styled.p>
          </div>
          <input
                    className="input__error-false"
                    name="name"
                    onChange={[Function]}
                    placeholder="place name here..."
                    required={true}
                    type="text"
                    value="John Doe" />
          <ul
                    className="input__errors">
                    
          </ul>
</label>,
        "_debugID": 2,
        "_renderedOutput": <label
          className="input">
          <div
                    className="input__label">
                    <styled.p>
                              name
                              <span
                                        className="input__required">
                                        •
                              </span>
                    </styled.p>
          </div>
          <input
                    className="input__error-false"
                    name="name"
                    onChange={[Function]}
                    placeholder="place name here..."
                    required={true}
                    type="text"
                    value="John Doe" />
          <ul
                    className="input__errors">
                    
          </ul>
</label>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown
    errors={Array []}
    isRequired={true}
    label="name"
    name="name"
    onChange={[Function]}
    placeholder="place name here..."
    type="text"
    value="John Doe" />,
}
`;
